### ISSUE TITLE
 self.enc_mask_token = nn.Parameter(torch.zeros(1, in_dim))

### ISSUE THREAD

**Issue by songshuhan** [THUDM/GraphMAE#34]
你好，我在看源码的时候看到
self.enc_mask_token = nn.Parameter(torch.zeros(1, in_dim))
....
out_x[token_nodes] += self.enc_mask_token

想咨询一下为什么所有的token_nodes在重建的时候全是共享的呢？这里self.enc_mask_token是不是应该是nn.Parameter(torch.zeros(num_token_nodes, in_dim))？？

**Comment by songshuhan** [comment]
这样重建出来的所有token_nodes的特征是全都一样的么？

**Comment by songshuhan** [comment]
想咨询一下共享向量的作用是什么？

**Comment by THINK2TRY** [comment]
@songshuhan mask_token [MASK] 的作用是指示对应的节点是被 mask 掉的，需要模型在训练的时候预测这个点的特征，所以不需要每个点有一个独特的 learnable vector

**Comment by songshuhan** [comment]
感谢作者的回复！我git clone了一下代码运行了一下，原始的代码在cora节点分类上的准确率在82.5%左右，然后我不加self.enc_mask_token,在cora上的准确率是81.5%左右，看起来加上确实会好一点。

不过每个节点都是根据encoder从邻居节点聚合信息来更新自己的embedding，而且我们也知道token_nodes的索引，好像并不需要一个“指示”向量加入才可以训练更新到被mask的节点，所以我感觉还是没太理解这个意义是什么？

**Comment by songshuhan** [comment]
感谢作者的回复，受益匪浅
